mixins/provider-mixin.js

```diff
@@ -0,0 +1,30 @@
+export const ProviderMixin = superclass => class extends superclass {
+	constructor() {
+		super();
+		this._instances = new Map();
+		this.addEventListener('d2l-request-instance', e => {
+			if (this._instances.has(e.detail.key)) {
```

Comment:
TBH, I'm not sure which is better, but this could be:
```
const instance = this._instances.get(e.detail.key);
if (instance) {
    e.detail.instance = instance;
    e.stopPropagation();
}
```
In the case where it exists, it looks like the browser will be doing more than one lookup.  To what extent, when it does not exist, does calling `has` help perf?

Reply:
One main difference is that in your example, if the map contains a key entry with a value of `false` (or `null` or `undefined`), it won't be returned.